src/ZatData.cpp

```diff
@@ -496,41 +242,42 @@ bool ZatData::InitSession(bool isReinit)
     }
   }
 
-  const Value& session = doc["session"];
+  const Value& account = doc["account"];
+  const Value& nonlive = doc["nonlive"];
 
-  m_countryCode = GetStringOrEmpty(session, "aliased_country_code");
-  m_serviceRegionCountry = GetStringOrEmpty(session, "service_region_country");
-  m_recallEnabled = session["recall_eligible"].GetBool();
-  m_selectiveRecallEnabled =
-      session.HasMember("selective_recall_eligible") ?
-          session["selective_recall_eligible"].GetBool() : false;
-  m_recordingEnabled = session["recording_eligible"].GetBool();
-  kodi::Log(ADDON_LOG_INFO, "Country code: %s", m_countryCode.c_str());
+  m_countryCode = Utils::JsonStringOrEmpty(doc, "current_country");
+  m_serviceRegionCountry = Utils::JsonStringOrEmpty(account, "service_country");
+  m_recallEnabled = Utils::JsonStringOrEmpty(nonlive, "replay_availability") == "available";
+  m_recordingEnabled = nonlive["recording_number_limit"].GetInt() > 0;
```

Comment:
This line has generated on test a assert fail with `kodi.bin: /home/alwin/Development/Kodi/_Addons/_alwin/pvr/pvr.zattoo/build/build/depends/include/rapidjson/document.h:1053: rapidjson::GenericValue<Encoding, Allocator>& rapidjson::GenericValue<Encoding, Allocator>::operator[](const rapidjson::GenericValue<Encoding, SourceAllocator>&) [with SourceAllocator = rapidjson::MemoryPoolAllocator<>; Encoding = rapidjson::UTF8<>; Allocator = rapidjson::MemoryPoolAllocator<>]: Assertion `false' failed.`

Here how I fixed it https://github.com/AlwinEsch/pvr.zattoo/commit/4621554cf4847e52620bcb234af52bd3faaea49b (the `set_target_properties` there on CMakeLists.txt is for another part, this prevent the `declared with greater visibility than the type of its field` warning on compile)

There are also other places with rapidjson on request where not checked, makes it sense to add checks there too?

Backtrace
```gdb
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff5a05859 in __GI_abort () at abort.c:79
#2  0x00007ffff5a05729 in __assert_fail_base
    (fmt=0x7ffff5b9b588 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", assertion=0x7fff84f757c8 "false", file=0x7fff84f75138 "/home/alwin/Development/Kodi/_Addons/_alwin/pvr/pvr.zattoo/build/build/depends/include/rapidjson/document.h", line=1053, function=<optimized out>) at assert.c:92
#3  0x00007ffff5a16f36 in __GI___assert_fail
    (assertion=0x7fff84f757c8 "false", file=0x7fff84f75138 "/home/alwin/Development/Kodi/_Addons/_alwin/pvr/pvr.zattoo/build/build/depends/include/rapidjson/document.h", line=1053, function=0x7fff84f756a0 "rapidjson::GenericValue<Encoding, Allocator>& rapidjson::GenericValue<Encoding, Allocator>::operator[](const rapidjson::GenericValue<Encoding, SourceAllocator>&) [with SourceAllocator = rapidjson::Mem"...) at assert.c:101
#4  0x00007fff84e36505 in rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator> >::operator[]<rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator> >(rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator> > const&) (this=0x7fff801a4cb0, name=...)
    at /home/alwin/Development/Kodi/_Addons/_alwin/pvr/pvr.zattoo/build/build/depends/include/rapidjson/document.h:1053
#5  0x00007fff84e3518b in rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator> >::operator[]<char const>(char const*)
    (this=0x7fff801a4cb0, name=0x7fff84f8d0cd "recording_number_limit") at /home/alwin/Development/Kodi/_Addons/_alwin/pvr/pvr.zattoo/build/build/depends/include/rapidjson/document.h:1033
#6  0x00007fff84e35ccb in rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator> >::operator[]<char const>(char const*) const
    (this=0x7fff801a4cb0, name=0x7fff84f8d0cd "recording_number_limit") at /home/alwin/Development/Kodi/_Addons/_alwin/pvr/pvr.zattoo/build/build/depends/include/rapidjson/document.h:1036
#7  0x00007fff84e4e9eb in ZatData::InitSession(bool) (this=0x7fff8001d290, isReinit=false) at /home/alwin/Development/Kodi/_Addons/_alwin/pvr/pvr.zattoo/src/ZatData.cpp:251
#8  0x00007fff84e507b9 in ZatData::Initialize() (this=0x7fff8001d290) at /home/alwin/Development/Kodi/_Addons/_alwin/pvr/pvr.zattoo/src/ZatData.cpp:478
#9  0x00007fff84e4aabb in CZattooTVAddon::CreateInstance(int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, void*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, void*&)
    (this=0x7fff80026250, instanceType=107, instanceID="0x7fff8000bab0", instance=0x7fff8000bc98, version="7.1.0", addonInstance=@0x7fff87ffe298: 0x7fff8001d290)
    at /home/alwin/Development/Kodi/_Addons/_alwin/pvr/pvr.zattoo/src/client.cpp:42
#10 0x00007fff84e4b586 in kodi::addon::CAddonBase::ADDONBASE_CreateInstance(int, char const*, void*, char const*, void**, void*)
    (instanceType=107, instanceID=0x7fff8000bad0 "0x7fff8000bab0", instance=0x7fff8000bc98, version=0x555559649edd "7.1.0", addonInstance=0x7fff87ffe298, parent=0x0)
    at /home/alwin/Development/Kodi/_Addons/_alwin/pvr/pvr.zattoo/build/build/depends/include/kodi/AddonBase.h:571
#11 0x0000555557bf7092 in ADDON::CAddonDll::CreateInstance(ADDON_TYPE, void const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, void*, void*) (this=
    0x7fff8000e9d0, instanceType=ADDON_INSTANCE_PVR, instanceClass=0x7fff8000bab0, instanceID="0x7fff8000bab0", instance=0x7fff8000bc98, parentInstance=0x0)
    at /home/alwin/Development/Kodi/kodi-Matrix/xbmc/addons/binary-addons/AddonDll.cpp:251
#12 0x0000555557bfdb53 in ADDON::IAddonInstanceHandler::CreateInstance(void*) (this=0x7fff8000bab0, instance=0x7fff8000bc98)
    at /home/alwin/Development/Kodi/kodi-Matrix/xbmc/addons/binary-addons/AddonInstanceHandler.cpp:80
#13 0x000055555806d528 in PVR::CPVRClient::Create(int) (this=0x7fff8000bab0, iClientId=724471594) at /home/alwin/Development/Kodi/kodi-Matrix/xbmc/pvr/addons/PVRClient.cpp:165
#14 0x00005555580966b7 in PVR::CPVRClients::UpdateAddons(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (this=0x55555b3e0260, changedAddonId="")
    at /home/alwin/Development/Kodi/kodi-Matrix/xbmc/pvr/addons/PVRClients.cpp:156
#15 0x0000555558095dc6 in PVR::CPVRClients::Start() (this=0x55555b3e0260) at /home/alwin/Development/Kodi/kodi-Matrix/xbmc/pvr/addons/PVRClients.cpp:64
#16 0x00005555580cfefc in PVR::CPVRManager::<lambda()>::operator()(void) const (__closure=0x55555cc86e40) at /home/alwin/Development/Kodi/kodi-Matrix/xbmc/pvr/PVRManager.cpp:345
#17 0x00005555580d3e1e in CLambdaJob<PVR::CPVRManager::Init()::<lambda()> >::DoWork(void) (this=0x55555cc86e30) at /home/alwin/Development/Kodi/kodi-Matrix/xbmc/utils/JobManager.h:39
#18 0x000055555775cd33 in CJobWorker::Process() (this=0x55555c9184d0) at /home/alwin/Development/Kodi/kodi-Matrix/xbmc/utils/JobManager.cpp:55
#19 0x0000555557838def in CThread::Action() (this=0x55555c9184d0) at /home/alwin/Development/Kodi/kodi-Matrix/xbmc/threads/Thread.cpp:282
#20 0x00005555578383e5 in CThread::<lambda(CThread*, std::promise<bool>)>::operator()(CThread *, std::promise<bool>) const (__closure=0x55555c8534e8, pThread=0x55555c9184d0, promise=...)
    at /home/alwin/Development/Kodi/kodi-Matrix/xbmc/threads/Thread.cpp:140
#21 0x00005555578398cc in std::__invoke_impl<void, CThread::Create(bool)::<lambda(CThread*, std::promise<bool>)>, CThread*, std::promise<bool> >(std::__invoke_other, CThread::<lambda(CThread*, std::promise<bool>)> &&) (__f=...) at /usr/include/c++/9/bits/invoke.h:60
#22 0x00005555578397e3 in std::__invoke<CThread::Create(bool)::<lambda(CThread*, std::promise<bool>)>, CThread*, std::promise<bool> >(CThread::<lambda(CThread*, std::promise<bool>)> &&) (__fn=...)
    at /usr/include/c++/9/bits/invoke.h:95
#23 0x0000555557839707 in std::thread::_Invoker<std::tuple<CThread::Create(bool)::<lambda(CThread*, std::promise<bool>)>, CThread*, std::promise<bool> > >::_M_invoke<0, 1, 2>(std::_Index_tuple<0, 1, 2>)
    (this=0x55555c8534e8) at /usr/include/c++/9/thread:244
#24 0x0000555557839691 in std::thread::_Invoker<std::tuple<CThread::Create(bool)::<lambda(CThread*, std::promise<bool>)>, CThread*, std::promise<bool> > >::operator()(void) (this=0x55555c8534e8)
    at /usr/include/c++/9/thread:251
#25 0x0000555557839666 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<CThread::Create(bool)::<lambda(CThread*, std::promise<bool>)>, CThread*, std::promise<bool> > > >::_M_run(void)
    (this=0x55555c8534e0) at /usr/include/c++/9/thread:195
#26 0x00007ffff5572d84 in  () at /lib/x86_64-linux-gnu/libstdc++.so.6
#27 0x00007ffff7f6c609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#28 0x00007ffff5b02293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

Reply:
Thanks for your input. I have included your commit. I hope this is ok, if not, just let me know.

The fields that are not checked for their availability should always be there. It seems that this has change for recording_number_limit in the new api.